pkg/kernel/mcp/ip.go

```diff
@@ -0,0 +1,77 @@
+package kernel
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/modelcontextprotocol/go-sdk/mcp"
+	"github.com/ovn-kubernetes/ovn-kubernetes-mcp/pkg/kernel/types"
+)
+
+// GetIPCommandOutput MCP handler for ip utility operations.
+// GetIPCommandOutput executes 'ip' utility commands on a node.
+// Requires ip utility in the debug container image.
+func (s *MCPServer) GetIPCommandOutput(ctx context.Context, req *mcp.CallToolRequest, in types.ListIPParams) (*mcp.CallToolResult, types.Result, error) {
+	ipCliAvailable := s.UtilityExists(ctx, req, in.Node, in.Image, "ip")
+	if !ipCliAvailable {
+		return nil, types.Result{}, fmt.Errorf("error while getting ip data: mentioned image does not have ip utility")
+	}
+
+	if err := ValidateIPCommand(in.Command); err != nil {
+		return nil, types.Result{}, fmt.Errorf("error while getting ip data: %w", err)
+	}
+	if in.FilterParameters != "" {
+		if err := validateFilterParameters(in.FilterParameters); err != nil {
+			return nil, types.Result{}, fmt.Errorf("error while getting ip data: %w", err)
+		}
+	}
+
+	cmd := newCommand("ip")
+	cmd.addIfNotEmpty(in.Options, in.Options)
+	cmd.add(strings.Fields(in.Command)...)
+	cmd.addIfNotEmpty(in.FilterParameters, strings.Fields(in.FilterParameters)...)
+
+	stdout, err := s.executeCommand(ctx, req, in.Node, in.Image, cmd.build())
+	if err != nil {
+		return nil, types.Result{}, fmt.Errorf("error while getting ip data: %w", err)
+	}
+	return nil, types.Result{Data: stdout}, nil
+}
+
+// ValidateIPCommand validates that the IP command is allowed.
+func ValidateIPCommand(ipCommand string) error {
+	if _, err := strconv.Atoi(ipCommand); err == nil {
+		return fmt.Errorf("invalid ip command: %s", ipCommand)
+	}
+	splitCommand := strings.Fields(strings.TrimSpace(ipCommand))
+	if len(splitCommand) < 2 {
+		return fmt.Errorf("invalid ip command: %s", ipCommand)
+	}
+	validIpCommand := map[string]bool{
+		"address":   true,
+		"link":      true,
+		"neighbour": true,
+		"netns":     true,
+		"route":     true,
+		"rule":      true,
+		"vrf":       true,
```

Comment:
can you also include `ip xfrm state` and `ip xfrm policy` commands ? it's useful for IPsec.